src/generic/srchctlg.cpp

```diff
@@ -222,6 +222,14 @@ class wxSearchButton : public wxControl
     void OnPaint(wxPaintEvent&)
     {
         wxPaintDC dc(this);
+
+        // Clear the background in case of a user bitmap with alpha channel
+        wxColour bg = m_search->GetBackgroundColour();
```

Comment:
This `void OnPaint(wxPaintEvent&)` is the `wxSearchButton`'s one not the `wxSearchCtrl`'s one. So we need to erase the background of `wxSearchButton` with the color of the `wxSearchCtrl`, and this is needed only when the `wxSearchButton` bitmap has alpha channel, so that the transparent area will get the color of the `wxSearchCtrl` which is drawn before it in the Z order.